build/templates/_converters.py.mako

```diff
@@ -29,6 +29,54 @@ def convert_timedelta_to_microseconds(value, library_type):
     return _convert_timedelta(value, library_type, 1000000)
 
 
+# This converter is not called from the normal codegen path for function. Instead it is
+# call from init and is a special case. Also, it just returns a string rather than a ctype object
+def convert_init_with_options_dictionary(value, encoding):
+    if type(value) is str:
+        init_with_options_string = value
+    else:
+        good_keys = {
+            'rangecheck': 'RangeCheck',
+            'queryinstrstatus': 'QueryInstrStatus',
+            'cache': 'Cache',
+            'simulate': 'Simulate',
+            'recordcoercions': 'RecordCoercions',
+            'interchangecheck': 'InterchangeCheck',
+            'driversetup': 'DriverSetup',
+            'range_check': 'RangeCheck',
+            'query_instr_status': 'QueryInstrStatus',
+            'record_coercions': 'RecordCoercions',
+            'interchange_check': 'InterchangeCheck',
+            'driver_setup': 'DriverSetup',
+        }
+        # First we validate that only allowed keys are in the incoming dictionary
```

Comment:
Is this validation needed? What if you just pass it along and let the driver do the validation?